foo_spider_monkey_panel/js_objects/fb_ui_selection_holder.cpp

```diff
@@ -89,11 +89,40 @@ void JsFbUiSelectionHolder::SetPlaylistTracking()
     holder_->set_playlist_tracking();
 }
 
-void JsFbUiSelectionHolder::SetSelection( JsFbMetadbHandleList* handles )
+const std::array<const GUID*, 7> guids = {
+    &contextmenu_item::caller_undefined,
+    &contextmenu_item::caller_active_playlist_selection,
+    &contextmenu_item::caller_active_playlist,
+    &contextmenu_item::caller_playlist_manager,
+    &contextmenu_item::caller_now_playing,
+    &contextmenu_item::caller_keyboard_shortcut_list,
+    &contextmenu_item::caller_media_library_viewer
+};
```

Comment:
This should be wrapped into a function that checks the index (see https://github.com/TheQwertiest/foo_spider_monkey_panel/blob/ab822e6508cd0861116525fcf68856c778f34709/foo_spider_monkey_panel/utils/art_helpers.cpp#L226 for example) and moved to the anonymous namespace.

Reply:
also though about it, did not like it neither
there is another reference to that array here:

https://github.com/TheQwertiest/foo_spider_monkey_panel/blob/ab822e6508cd0861116525fcf68856c778f34709/foo_spider_monkey_panel/js_objects/fb_utils.cpp#L552-L570

I wonder where would that function be placed so both functions can call it... utils\menu_helpers.h ?

https://github.com/TheQwertiest/foo_spider_monkey_panel/blob/ab822e6508cd0861116525fcf68856c778f34709/foo_spider_monkey_panel/utils/menu_helpers.h#L1

The namespace there is smp::utils

what would you recommend? Just wrap it before SetSelection() or make it available to GetSelection()

Reply:
Nice catch =)

> I wonder where would that function be placed so both functions can call it... utils\menu_helpers.h ?

Just create new files in the root, i.e. `\foo_spider_monkey_panel\foo_spider_monkey_panel\context_menu_caller.(h/cpp)` with `namespace smp`. I have plans to reorganize some source files anyway. So will clean it up later.

Reply:
Well, oops...
The whole block might be as well be replaced by (my bad, sorry)
const GUID type = ui_selection_manager_v2::get()->get_selection_type( 0 );
and can maybe let GetSelectionType() alone as its doing the opposite conversion

I'll revise after having some rest. Also will check the changes in the documentation...
Thanks 
Best regards